apps/ios/SimpleX SE/ShareViewController.swift

```diff
@@ -32,3 +32,22 @@ class ShareViewController: UIHostingController<ShareView> {
         }
     }
 }
+
+// TODO: Database access management
+/// Handle database access using existing (not-ideal) model, since
+///
+/// 1. Subscribe to lifecycle notifications:
+///     - `NSExtensionHostDidBecomeActive`
+///     - `NSExtensionHostWillResignActive`
+///     - `NSExtensionHostDidEnterBackground`
+///     - `NSExtensionHostWillEnterForeground`
+///
+/// 2. Add `SEState` using `SharedFileSubscriber`
+/// and suspend the chat in case there is a NSE Notification, while share sheet process is active.
+///
+/// 3. Handle the case, where share-sheet and app is open simultaneously by sending
+/// a notification over `CFNotificationCenterGetDarwinNotifyCenter` 
+/// so the main app can register database changes and update it's `ChatModel`
```

Comment:
You can open share sheet form the app itself for example.
Also I can happen on iPad multitasking

Reply:
right... How would it work then? do we need to suspend the app in this case? Want the app be in background anyway in this case, so suspended? or it would go into .inactive state?